wagtailio/core/models/pages.py

```diff
@@ -130,3 +138,165 @@ class ContentPage(Page, HeroMixin, SocialMediaMixin, CrossPageMixin):
+        index.SearchField("introduction"),
+    ]
+
+    def _is_htmx_request(self, request):
+        return request.headers.get("HX-Request") == "true"
+
+    def _filter_used_sectors(self):
+        return ["All"] + list(
+            self.showcase_items.values_list("sector__name", flat=True).distinct()
+        )
+
+    def _filtered_showcase_items(self, sector):
+        if sector in self._filter_used_sectors() and sector != "All":
+            return self.showcase_items.filter(sector__name=sector)
+        else:
+            return self.showcase_items.all()
+
+    def get_template(self, request, *args, **kwargs):
+        if self._is_htmx_request(request):
+            return self.htmx_template
+        else:
+            return self.template
+
+    def serve(self, request):
+        response = super().serve(request)
+        if self._is_htmx_request(request):
+            # We only return the fragment, rather than the whole page.
+            new_url = self.url + "?" + request.GET.urlencode()
+            response.headers["HX-Push"] = new_url
+        return response
+
+    def get_context(self, request):
+        context = super().get_context(request)
+        sectors = {}
+
+        for sector in self._filter_used_sectors():
+            if sector == request.GET.get("sector"):
+                sectors[sector] = "selected"
+            else:
+                sectors[sector] = ""
+
+        context |= {
+            "sectors": sectors,
+            "showcase_items": self._filtered_showcase_items(request.GET.get("sector")),
+            "current_sector": request.GET.get("sector") or "All",
+        }
+
+        return context
+
+
+class ShowcaseItem(Orderable):
+    title = models.CharField(max_length=255)
+    description = models.TextField(blank=True)
+    image = models.ForeignKey(
+        "images.WagtailIOImage",
+        models.SET_NULL,
+        null=True,
+        related_name="+",
+    )
+    alt_text = models.CharField(max_length=255, blank=True)
+    logo = models.ForeignKey(
+        "images.WagtailIOImage",
+        models.SET_NULL,
+        null=True,
+        blank=True,
+        related_name="+",
+    )
+    external_link = models.URLField(blank=True)
+    internal_link = models.ForeignKey(
+        "wagtailcore.Page",
+        models.SET_NULL,
+        null=True,
+        blank=True,
+        related_name="+",
+    )
+    link_text = models.CharField(max_length=255)
+
+    sector = models.ForeignKey(
+        "Sector",
+        models.CASCADE,
+        null=False,
+        blank=False,
+        related_name="+",
+    )
+
+    page = ParentalKey(
+        ShowcasePage,
+        related_name="showcase_items",
+        on_delete=models.CASCADE,
+    )
+
+    panels = [
+        FieldPanel("title"),
+        FieldPanel("sector"),
+        FieldPanel("description"),
+        FieldPanel("image"),
+        FieldPanel("alt_text"),
+        FieldPanel("logo"),
+        FieldPanel("external_link"),
```

Comment:
suggestion: wrap the link panels in a `MultiFieldPanel` with some help text around what is required (i.e. you need to enter either an external or internal link). It is the little tweaks that lead to editorial delights.

also, it may be nice if you specified icons for the panels

Reply:
I've wrapped this in a `MultiFieldPanel` panel, although this site is still 4.2 so we cannot show icons until 5x :).